src/jaxoplanet/experimental/starry/light_curves.py

```diff
@@ -1,68 +1,88 @@
-    central_bodies_lc = jax.vmap(central_body_lc, in_axes=(None, 0, 0, 0, 0, None))
-    central_light_curve = (
-        central_bodies_lc(
-            system.central.map,
-            (system.radius / central_radius).magnitude,
-            (xos / central_radius).magnitude,
-            (yos / central_radius).magnitude,
-            (zos / central_radius).magnitude,
-            theta,
-        )
-        * system.central.map.amplitude
-    )
-
-    @partial(system.body_vmap, in_axes=(0, 0, 0))
-    def bodies_lc(body, x, y, z):
-        theta = time * 2 * jnp.pi / body.map.period
+    @partial(system.surface_map_vmap, in_axes=(0, 0, 0, 0, None))
+    def compute_body_light_curve(surface_map, radius, x, y, z, time):
+        theta = time.magnitude * 2 * jnp.pi / surface_map.period
         return (
-            jax.vmap(map_light_curve, in_axes=(None, None, 0, 0, 0, 0))(
-                body.map,
-                (central_radius / body.radius).magnitude,
-                (x / body.radius).magnitude,
-                (y / body.radius).magnitude,
-                (z / body.radius).magnitude,
+            map_light_curve(
+                surface_map,
+                (system.central.radius / radius).magnitude,
+                (x / radius).magnitude,
+                (y / radius).magnitude,
+                (z / radius).magnitude,
                 theta,
             )
-            * body.map.amplitude
+            * surface_map.amplitude
         )
 
-    bodies_light_curves = bodies_lc(-xos, -yos, -zos)
-    return jnp.vstack([central_light_curve, bodies_light_curves])
+    @quantity_input(time=ureg.day)
+    @vectorize
+    def light_curve_impl(time: Quantity) -> Array:
+        xos, yos, zos = system.relative_position(time)
+
+        if system.surface_map is None:
+            central_light_curves = None
+        else:
+            theta = time.magnitude * 2 * jnp.pi / system.surface_map.period
+            central_radius = system.central.radius
+            central_light_curves = (
+                central_bodies_lc(
+                    system.surface_map,
+                    (system.radius / central_radius).magnitude,
+                    (xos / central_radius).magnitude,
+                    (yos / central_radius).magnitude,
+                    (zos / central_radius).magnitude,
+                    theta,
+                )
+                * system.surface_map.amplitude
+            )
+
+        if all(surface_map is None for surface_map in system.surface_maps):
+            body_light_curves = None
+        else:
+            body_light_curves = compute_body_light_curve(  # type: ignore
+                system.radius, -xos, -yos, -zos, time
+            )
+
+        result = jnp.zeros(system.shape, dtype=time.dtype)
+        if central_light_curves is not None:
+            result += central_light_curves
+        if body_light_curves is not None:
+            result += body_light_curves
+
+        return result
```

Comment:
My remaining question here is whether or not we'd ever want to expose the central and body light curves separately or does it make sense to always add them together?

Reply:
Let me rephrase the question: _Why should we prevent a curious user from learning about the decomposition of the light curve into the individual contributions of the various bodies in the system?_
Perhaps we could add a keyword to return the sum of all light curves instead of the individual light curves (or vice versa, depending on what kind of default behaviour we think would be the more intuitive).

PS. If I understand you correctly, I should open another PR to pull from `fix-134` in my fork to `fix-134` in this repository.

Reply:
> Perhaps we could add a keyword to return the sum of all light curves instead of the individual light curves (or vice versa, depending on what kind of default behaviour we think would be the more intuitive).

I think that's exactly the question!

> PS. If I understand you correctly, I should open another PR to pull from fix-134 in my fork to fix-134 in this repository.

Yes please!

Reply:
Starry's default behavior is to return the total flux by default--I think that would make sense here too

https://starry.readthedocs.io/en/latest/System/#starry.System.flux

Reply:
I agree. Especially since getting an array instead is a more natural result compared to a tupel containing both `None` values and arrays. So I think we should also set something like `total=False` by default.

Reply:
@shashankdholakia — The `total` here is actually a little different! In `jaxoplanet` our tradition is to always return separate arrays for each body. The question here is how we should handle the fact that there are separate light curves for the central and orbiting bodies. Do we want to sum the central and the orbiting bodies? Or have an array that has a shape of `1 + n_bodies`.

Reply:
In that case, I'd prefer to have `1+n_bodies` with the central prepended to the front since it could be useful to model phase curves of the planets individually

Reply:
A `total` keyword is easily replaceable by a `.sum` with a fine control over the axes behind summed (thinking about multi-wavelength maps here). So I think I would personally prefer to see a different behavior than starry here. Although that's only minor

Reply:
Great! In that case, I think that the right thing to return would be a tuple with `(central_light_curve, body_light_curves)` since these are qualitatively different objects. `central_light_curve` would have shape `(num_times,)` and `body_light_curves` would have shape `(num_bodies, num_times)`.

Then we can add a keyword argument (like `total`) to decide if we want everything to be summed.

Reply:
What about the case where one of the orbiting bodies has no surface map, i.e. surface maps were only specified for some but not all of the orbiting bodies? Will its whole column in `body_light_curves` just be filled with 0 or `NaN` values?

Reply:
I agree that returning a `1+n_bodies` shape array (current) would play nice here

Reply:
I'm inclined to move this conversation to a new issue (@lgrcia and I have identified another related problem to talk about), and get this PR merged since @dgegen has fixed the remaining bug and the results are consistent with what we had previously. I'll tag everyone over in the new issue!